src/main/java/com/redhat/lightblue/healthcheck/HealthCheckResource.java

```diff
@@ -44,28 +47,81 @@
 
     private static final String ENTITY = "test";
 
-    private final LightblueClient client;
+    private final Map<String, LightblueClient> lightblueClients;
     private final String hostname;
 
     public HealthCheckResource() throws Exception {
-        client = new LightblueHttpClient();
+
+        lightblueClients = new LinkedHashMap<>();
+
+        //this config file contains one line for each lightblue client properties file that
+        // should be read in and used.
+        InputStream stream = this.getClass().getResourceAsStream("lightblue-clients.config");
+        if(null != stream) {
+            BufferedReader reader = new BufferedReader(new InputStreamReader(stream));
+
+            String clientConfigFilePath;
+            while ((clientConfigFilePath = reader.readLine()) != null) {
+                lightblueClients.put(
+                        clientConfigFilePath.replace(".properties", ""),
+                        new LightblueHttpClient(clientConfigFilePath)
+                );
+            }
+        } else {
+            lightblueClients.put("lightblue-client", new LightblueHttpClient());
+        }
+
         hostname = InetAddress.getLocalHost().getHostName();
     }
 
     @GET
     @Path("/health")
     public Response health() {
+
+        HealthCheck healthCheck = new HealthCheck();
+
+        for (Map.Entry<String,LightblueClient> client :  lightblueClients.entrySet()) {
+
+            HealthCheckStatus healthCheckStatus = checkHealth(client);
+
+            if(HealthCheckStatus.Status.error.name().equals(healthCheckStatus.getStatus().name())) {
+                healthCheck.hasFailures(true);
+            }
+
+            healthCheck.addStatus(client.getKey(), checkHealth(client));
+        }
+
+        if (healthCheck.hasFailures()) {
+            StringBuffer errorMessages = new StringBuffer();
+            for(Map.Entry<String,HealthCheckStatus> healthCheckStatus : healthCheck.getClientStatuses().entrySet()) {
+                if(HealthCheckStatus.Status.error.name().equals(healthCheckStatus.getValue().getStatus().name())) {
+                    errorMessages.append(healthCheckStatus.getValue().getMessage());
+                }
+            }
+            return Response.status(javax.ws.rs.core.Response.Status.INTERNAL_SERVER_ERROR).entity(
+                    "{[" + errorMessages.toString() + "]}"
+            ).build();
+        } else {
+            return Response.status(javax.ws.rs.core.Response.Status.OK).entity(
+                "{\"status\":\"success\"}"
+            ).build();
+        }
+
+    }
+
+    private HealthCheckStatus checkHealth(Map.Entry<String,LightblueClient> client) {
```

Comment:
It might be easier to read if you just passed in the LightblueClient and a name for it.